### PR TITLE
Feature/3286 example for drawer nav item

### PR DESCRIPTION
**Pull request by ektasawant**
Fixes 3286 .

#### Changes proposed in this Pull Request:
- Add examples for drawer nav item
- I have added examples similar to angular dev docs. Let me know do I need to add more.

#### To Test:
- http://localhost:3000/components/drawer/drawer-nav-item/examples

Note: **styling for mdx/heading/ will be part of https://jira-prod.tcc.etn.com/browse/BLUI-3323**

### CODE REVIEW

File: docs/src/components/componentList/drawerNavItemComponent/BasicDrawerNavItem.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import * as colors from '@brightlayer-ui/colors';
+
+const htmlCode = `<DrawerBody>
+  <DrawerNavGroup>
+    <DrawerNavItem title="Dashboard" icon={<Dashboard/>} itemID="1" />
+    <DrawerNavItem title="Locations" icon={<Notifications />} itemID="2" />
+    <DrawerNavItem title="Legal" icon={<Gavel />} itemID="3" />
+  </DrawerNavGroup>
+</DrawerBody>
+`;
```
Comment (daileytj) [inline docs/src/components/componentList/drawerNavItemComponent/BasicDrawerNavItem.tsx:14]: The rendered code doesn't match the copy/paste code. Notably the styles and the icons.

**Comment by daileytj** [comment]
Is the "copy code" button and the "full code on github" link within scope of this story?

**Comment by ektasawant** [comment]
> https://blui-react-docs--pr467-feature-3286-example-napcdlkf.web.app

Thanks for observing, will remove icon related code from copy code. In this story copy code and syntax highlighting, github link is not scope.

File: docs/src/components/componentList/drawerNavItemComponent/DrawerNavItemExamples.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
```
Comment (bkarambe) [inline docs/src/components/componentList/drawerNavItemComponent/DrawerNavItemExamples.tsx:1]: There is no need of this component file. You can directly import Markdown file inside the navigation and use it.

**Review by bkarambe — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts.

File: docs/src/components/componentList/drawerNavItemComponent/BasicDrawerNavItem.tsx (hunk 1 above)
Comment (ektasawant) [reply docs/src/components/componentList/drawerNavItemComponent/BasicDrawerNavItem.tsx]: done please check

**Review by huayunh — COMMENTED** [review]
- Dividers shall be off by default
- Can we enable "line highlighting" from prismjs
- I don't see any "round highlight" here
image
- Why is syntax highlighting failing?
image
- Your nested drawer items overflowed the drawer height. I expect, in this case, that the drawer expand to accommodate the increased height
image

**Comment by ektasawant** [comment]
> * Dividers shall be off by default - **Which divider ?**
> * Can we enable "line highlighting" from prismjs - **This is not part of the current story. Will fix this in another story**
> * I don't see any "round highlight" here - **fixed**
>   image
> * Why is syntax highlighting failing? - **fixed**
>   image
> * Your nested drawer items overflowed the drawer height. I expect, in this case, that the drawer expand to accommodate the increased height - **without height it doesn't work** 
>   image

**Comment by huayunh** [comment]
Screen Shot 2022-08-01 at 9 39 20 AM

**Comment by huayunh** [comment]
In "Drawer Nav Items With Icons", can you keep the code indentation to be consistent (always 4 spaces)?

**Comment by huayunh** [comment]
There's a z-index problem here (I guess) 

https://user-images.githubusercontent.com/8997218/182162216-2d9fa9f5-f5bc-4ccb-af9d-66c57051b4c5.mov

**Comment by ektasawant** [comment]
> Screen Shot 2022-08-01 at 9 39 20 AM

mdx styling will be part of https://jira-prod.tcc.etn.com/browse/BLUI-3323

File: docs/src/markdown/examples/DrawerNavItemExamples.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# Nested Drawer Nav Items
+
+Navigation items can be nested to allow for different levels of navigation. To create a nested navigation item, simply place a `<DrawerNavItem />` inside of another. The `<DrawerNavItem />` allows nested navigation up to 3 levels deep.
+
+<DrawerNavItemNestedItems />
+
+# Selected Navigation Items
+
+activeItem prop of Drawer should be updated with itemID of selected `<DrawerNavItem />`. The active item's highlighted shape can be adjusted via the [activeItemBackgroundShape] prop to be either "square" (default) or "round".
```
Comment (huayunh) [inline docs/src/markdown/examples/DrawerNavItemExamples.mdx:24]: When I suggested "monospace", I meant these...

```suggestion
`activeItem` prop of Drawer should be updated with `itemID` of selected `<DrawerNavItem />`. The active item's highlighted shape can be adjusted via the `activeItemBackgroundShape` prop to be either "square" (default) or "round".
```

**Comment by huayunh** [comment]
Everything else looks good.

**Review by bkarambe — APPROVED** [review]
Looks good

**Pull request merged**